### PR TITLE
Test memory constaints

### LINKED ISSUE #655: integration/docker: add test to check memory constraints inside the VM
use next docker options to run containers with memory constraints

```
  -m, --memory bytes                   Memory limit
      --memory-reservation bytes       Memory soft limit
      --memory-swap bytes              Swap limit equal to memory plus swap: '-1' to enable unlimited swap
      --memory-swappiness int          Tune container memory swappiness (0 to 100) (default -1)
```

see https://github.com/kata-containers/runtime/issues/653

### PR DESCRIPTION
**Pull request by devimc**
run a container, update its memory constraints and check if the
constraints are applied in the memory cgroup

fixes #655

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by GabyCT** [comment]
lgtm

File: integration/docker/mem_test.go [hunk 1]
```diff
@@ -37,4 +48,74 @@ var _ = Describe("memory constraints", func() {
 			Expect(stderr).To(ContainSubstring("fatal error: runtime: out of memory"))
 		})
 	})
+
+	Context("run container using memory constraints", func() {
+		It("should have applied the constraints", func() {
+			// 512MB
+			memSize = "536870912"
```
Comment (jodh-intel) [inline integration/docker/mem_test.go]: Nit: Might be worth making this "self-documenting". Something like:

```
mb := 1024 * 1024
memSize = 512 * mb
```

**Comment by jodh-intel** [comment]
lgtm

File: integration/docker/mem_test.go (hunk 1 above)
Comment (devimc) [reply integration/docker/mem_test.go]: done

**Comment by jodh-intel** [comment]
Thanks.

**Comment by chavafg** [comment]
lgtm

**Pull request merged**